mobius/src/geometry/line_segment.rs

```diff
@@ -12,6 +12,18 @@ impl LineSegment {
     pub fn new(start: Complex, end: Complex) -> Self {
         Self { start, end }
     }
+
+    pub fn reverse(&self) -> Self {
+        let &Self { start, end } = self;
+        Self {
+            start: end,
+            end: start,
+        }
+    }
+
+    pub fn interpolate(&self, t: f64) -> Complex {
```

Comment:
If this were added to `DirectedEdge`, the value would have to be an `Option<Complex>` at best, or a `Result<Complex, Error>` as there are some cases where this doesn't make sense. E.g. t out of bounds isn't always defined like it is for a line. If one of the endpoints are infinite, there isn't a clear mapping from t values to points on the segment in general...

For now, leave it as is.